### DISCUSSION TITLE
✅ 🔗 Conditional / Dynamic Visibility (Duplicate of issue #4246)

### DISCUSSION THREAD

**Discussion by natashdev** [elementor/elementor#21152] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Please Add Something Like Dynamic Visibility for Widgets and contents or something like this. we want to make something visible on condition , like (make it invisible if empty). if we can hide some content if empty that will be so helpful,
Thanks

### Describe the solution you'd like

Please Add Something Like Dynamic Visibility for Widgets and contents or something like this. we want to make something visible on condition , like (make it invisible if empty),
if we can hide some content if the dynamic content is empty, that will be so helpful.
Thanks

**Comment by Michaelgimii** [comment]
In my opinion, this is a very necessary feature that should be prioritized. It has been almost 7 years since Elementor was born. Elementor is incredibly powerful and very popular, but it lacks such an important feature. This feature will definitely be used in any website. in different situations. Therefore, currently Elementor users must depend on third-party plugins to meet this requirement. Plugins that may not meet the required standards.
I hope Elementor has a plan to add this feature. I can confidently say that every user will vote positively for this feature. Just create a survey.

**Comment by black-eye** [comment]
This is ablolutely essential! Unfortunately, it's not even on the roadmap. And you don't have to think twice, wheather it's requested or not, it's quite clear - it's the most requested/commented issue on GitHub!

Not to mention there are several addons able to deliver this funcionality even without touching the core. So how difficult it could be to do it natively?

**Comment by fabigrafo** [comment]
It’s the most wanted on github, i think yes they are work on this. The question is when they release it.

![EA85DF50-1EE6-43F1-AF9A-215570289809](https://user-images.githubusercontent.com/57997262/216327110-1b21f649-ba31-445c-b042-280629e0ec2b.png)

**Comment by Michaelgimii** [comment]
Last year @avivu said they are working on it. But no time is known for its release. I think this is very important. Even more important than Grid Container!

**Reply by fabigrafo** [reply]
> Last year @avivu said they are working on it. But no time is known for its release. I think this is very important. Even more important than Grid Container!

Both are important. I think Grid Container is CSS Grid?

**Reply by Michaelgimii** [reply]
Yes, it is. Elementor calls it Container Grid. In fact, this is how it is introduced in the roadmap.

**Comment by 79ho3ein** [comment]
This feature is very important and useful. I am currently using the elementor exras plugin to meet this need and unfortunately this plugin no longer releases any updates. So I have to think of an alternative for it. I hope Elementor fills this important gap for users.
This is definitely the most popular feature among Elementor users.

**Comment by Joliamendi** [comment]
I wish the developers would tell us some information about this feature. I appreciate it. Even if the release is for another year or the end of this year.

**Comment by fabigrafo** [comment]
Hi @nicholaszein can you tell us more about this feature? We need this! 👀

**Comment by nicholaszein** [comment]
Hi everyone,

Thank you for your feedback! 🙏 

### 🔗 🚀 There is already a similar request submitted by someone here #4246.

I am currently in the process of migrating that request to a new discussion.

For those requesting updates on this feature and when it will be released, know that we don't have a release date yet. There are still plenty of structural changes we need to do before we are ready to release this feature.

We ask you to be patient. We will add this feature, but not in the near future.

Kind regards,

Nick

**Reply by nicholaszein** [reply]
For now, I will be closing this Request to avoid duplicates. When the time comes to migrate that issue here, I'll let you all know.

@black-eye, I have a little request for you. Please if you will, do not direct people to this request or any other duplicate request here. 


Thanks 🙏

**Comment by nicholaszein** [comment]
Hello!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0, and improved with Elementor Pro 3.20.0! 🥳 

You can now use Display Conditions for widgets based on a variety of conditions, including date, time, user role, logged in/out, and others (more to come too). Check the [documentation for the new Display Conditions](https://elementor.com/help/show-and-hide-elements-on-a-page-display-conditions/) for widgets (Pro).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
